widgets/search-box/__tests__/search-box-test.js

```diff
@@ -168,6 +168,59 @@ describe('search-box()', () => {
     });
   });
 
+  context('input', () => {
+    let input;
+    beforeEach(() => {
+      container = document.createElement('div');
+      input = createHTMLNodeFromString('<input />');
+      input.addEventListener = sinon.spy();
+    });
+
+    context('instant search', () => {
+      beforeEach(() => {
+        widget = searchBox({container, autofocus: 'auto'});
+        widget.getInput = sinon.stub().returns(input);
+      });
+
+      it('do not perform the search on keyup', () => {
+        // Given
+        helper.state.query = 'foo';
+        // When
+        widget.init(initialState, helper);
+        // Then
+        expect(input.addEventListener.called).toEqual(true);
+        expect(input.addEventListener.args[1].length).toEqual(3);
+        expect(input.addEventListener.args[1][0]).toEqual('input');
+        let fn = input.addEventListener.args[1][1];
+        fn({currentTarget: {value: 'test'}});
+        expect(helper.setQuery.calledOnce).toBe(true);
+        expect(helper.search.called).toBe(true);
+      });
+    });
+
+    context('non-instant search', () => {
+      beforeEach(() => {
+        widget = searchBox({container, autofocus: 'auto', searchOnEnterKeyPressOnly: true});
+        widget.getInput = sinon.stub().returns(input);
+      });
+
+      it('does not performs the search on input event', () => {
+        // Given
+        helper.state.query = 'foo';
+        // When
+        widget.init(initialState, helper);
+        // Then
+        expect(input.addEventListener.called).toEqual(true);
+        expect(input.addEventListener.args[1].length).toEqual(3);
+        expect(input.addEventListener.args[1][0]).toEqual('input');
+        let fn = input.addEventListener.args[1][1];
+        fn({currentTarget: {value: 'test'}});
+        expect(helper.setQuery.calledOnce).toBe(true);
+        expect(helper.search.called).toBe(false);
```

Comment:
The only difference between the two tests is that `helper.search` is not called when `searchOnEnterKeyPressOnly: true`, still the test is very long. Shouldn't we just do a smaller test for that?

I find this two tests hard to follow